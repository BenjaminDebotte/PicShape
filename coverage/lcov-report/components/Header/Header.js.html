<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Header/Header.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">components/Header</a> Header.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/8</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { IndexLink, Link } from 'react-router';
import { connect } from 'react-redux'
import { logout } from '../../actions/auth';
&nbsp;
import './Header.css';
&nbsp;
export class Header extends Component {
&nbsp;
<span class="fstat-no" title="function not covered" >    ha</span>ndleLogout(event) {
<span class="cstat-no" title="statement not covered" >      event.preventDefault();</span>
<span class="cstat-no" title="statement not covered" >      this.props.dispatch(logout());</span>
    }
&nbsp;
    render() {
        const active = { borderBottomColor: '#93bf41' };
&nbsp;
        const unloggedComponents = ([
            &lt;Link className="item" to="/login" activeStyle={active}&gt;Login&lt;/Link&gt;,
            &lt;Link className="item" to="/signup" activeStyle={active}&gt;Signup&lt;/Link&gt;
        ]);
&nbsp;
        const loggedComponents = ([
            &lt;Link className="item" to="/account"&gt;My Account&lt;/Link&gt;,
            &lt;Link className="item" onClick={this.handleLogout.bind(this)}&gt;Logout&lt;/Link&gt;
        ]);
&nbsp;
      return (
          &lt;div className="ui inverted menu"&gt;
            &lt;div className="ui container"&gt;
                &lt;Link className="header item" to="/home"&gt;
                &lt;img className="logo logo-spin" src="logo.ico"/&gt;
                      Picshape
                  &lt;/Link&gt;
                &lt;Link className="item" to="/upload" activeStyle={active}&gt;Upload&lt;/Link&gt;
                { this.props.token ? <span class="branch-0 cbranch-no" title="branch not covered" >loggedComponents </span>: unloggedComponents }
                &lt;Link className="item" to="/about" activeStyle={active}&gt;About&lt;/Link&gt;
            &lt;/div&gt;
          &lt;/div&gt;
&nbsp;
      );
    }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    token: state.auth.token,
    user: state.auth.user
  };
};
&nbsp;
export default connect(mapStateToProps)(Header);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Thu Mar 09 2017 10:50:14 GMT+0100 (Romance Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
