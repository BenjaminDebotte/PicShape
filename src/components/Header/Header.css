.main.container {
  margin-top: 2em;
}

.ui.menu .item img.logo {
  margin-right: 1.5em;
}
