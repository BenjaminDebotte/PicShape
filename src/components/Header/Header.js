import React, { Component } from 'react';
import { IndexLink, Link } from 'react-router';
import { connect } from 'react-redux'
import { logout } from '../../actions/auth';

import './Header.css';

class Header extends Component {

    handleLogout(event) {
      event.preventDefault();
      this.props.dispatch(logout());
    }

    render() {
        const active = { borderBottomColor: '#93bf41' };

        const unloggedComponents = ([
            <Link className="item" to="/login" activeStyle={active}>Login</Link>,
            <Link className="item" to="/signup" activeStyle={active}>Signup</Link>
        ]);

        const loggedComponents = ([
            <Link className="item" to="/wall">Wall</Link>,
            <Link className="item" to="/gallery">My Gallery</Link>,
            <Link className="item" to="/account">My Account</Link>,
            <Link className="item" onClick={this.handleLogout.bind(this)}>Logout</Link>
        ]);

      return (
          <div className="ui inverted menu">
            <div className="ui container">
                <Link className="header item" to="/home">
                <img className="logo logo-spin" src="logo.ico"/>
                      Picshape
                  </Link>
                <Link className="item" to="/upload" activeStyle={active}>Upload</Link>
                { this.props.token ? loggedComponents : unloggedComponents }
                <Link className="item" to="/about" activeStyle={active}>About</Link>
            </div>
          </div>

      );
    }
    }


    const mapStateToProps = (state) => {
      return {
        token: state.auth.token,
        user: state.auth.user
      };
    };

    export default connect(mapStateToProps)(Header);
