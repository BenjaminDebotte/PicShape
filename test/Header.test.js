import React from 'react';
import  {Header} from '../src/components/Header/Header';
import renderer from 'react-test-renderer';

describe('components', function() {
  describe('<Header />', function() {
    it('renders correctly', function() {
      var tree = renderer.create(<Header />).toJSON();
      expect(tree).toMatchSnapshot();
        });
  });
});
