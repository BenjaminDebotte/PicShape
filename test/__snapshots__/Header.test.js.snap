exports[`components <Header /> renders correctly 1`] = `
<div
  className="ui inverted menu">
  <div
    className="ui container">
    <a
      className="header item"
      onClick={[Function bound handleClick]}
      style={Object {}}>
      <img
        className="logo logo-spin"
        src="logo.ico" />
      Picshape
    </a>
    <a
      className="item"
      onClick={[Function bound handleClick]}
      style={Object {}}>
      Upload
    </a>
    <a
      className="item"
      onClick={[Function bound handleClick]}
      style={Object {}}>
      Login
    </a>
    <a
      className="item"
      onClick={[Function bound handleClick]}
      style={Object {}}>
      Signup
    </a>
    <a
      className="item"
      onClick={[Function bound handleClick]}
      style={Object {}}>
      About
    </a>
  </div>
</div>
`;
